core/src/main/scala/org/apache/spark/HttpServer.scala

```diff
@@ -155,6 +158,7 @@ private[spark] class HttpServer(
       throw new ServerStateException("Server is already stopped")
     } else {
       server.stop()
+      condOpt(server.getThreadPool) { case x: LifeCycle => x }.foreach(_.stop())
```

Comment:
I don't think we are going for idiomatic Scala in Spark. Actually in most cases we go for readability.